packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -1373,6 +1438,47 @@ function generateExportName(stackExports: Construct, id: string) {
   return prefix + localPart.slice(Math.max(0, localPart.length - maxLength + prefix.length));
 }
 
+function isExportAList(exportedValue: any, resolved: any) {
+  const target = exportedValue.target;
+  const resolvedGetAtt = resolved['Fn::GetAtt'];
+  const resolvedRef = (resolved['Fn::Ref'] ?? resolved.Ref);
+
+  if (resolvedGetAtt && resolvedRef) {
+    throw new Error(`found a reference with both 'Fn::GetAtt' and 'Fn::Ref' ${resolved}`);
```

Comment:
This will pretty likely say `[object Object]`. You'll need to `JSON.stringify()` it.